### ISSUE TITLE
Change react-sticky dependency (contains deprecated code)

### ISSUE THREAD

**Issue by albertms10** [ant-design/ant-design#24775]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
When using `react-sticky`’s components, a warning is thrown with the following message:
```
Warning: componentWillMount has been renamed, and is not recommended for use.
See https://fb.me/react-unsafe-component-lifecycles for details.
…
Please update the following components: Sticky
```
Knowing the component hasn’t been updated in 2 years, it may be a good decision to move forward and use another similar package.

### What does the proposed API look like?
A similar dependency that has been updated is [`react-stickynode`](https://www.npmjs.com/package/react-stickynode) and works similarly, without any warning.

However, if a `Tabs` component is placed inside it, it won’t show its ink bar (as happened in Ant Design v4.3.0 with `react-sticky`).
If you try to replace it with the native `Affix` component, it behaves the same (no ink bar).

Please, consider updating the `Sticky`-related dependency or just fix the `Tab` usage inside an `Affix` component (to achieve the same result natively).

**Comment by afc163** [comment]
@bralemili @albertms10 Could you help to update demo?

**Comment by bralemili** [comment]
I used react-stickynode in the example and it does not fix the problem mentioned in #28646
I will gladly updated the demo once we have a working solution

**Comment by aashayamballi** [comment]
In our case we're using one div to fix at the top using affix and we're using react-stickynode to stack the tab header to  div sticky component when scrolled. we're uanble to make it work. is there any solution for this? @afc163 or @bralemili do you have any solutions for this?
